example/src/screens/MainScreen.tsx

```diff
@@ -74,11 +99,76 @@ export default class MainScreen extends Component<any, MainScreenState> {
       .then((response) => response.data);
   }
 
+  componentDidMount() {
+    this.updateTrackingConsent()
+  }
+
+  updateTrackingConsent() {
+    getTrackingConsent().then(consent => {
+      this.setState({
+        trackingConsent: consent
+      })
+    })
+  }
+
+  setTrackingConsentModalVisible(visible: boolean) {
+    this.setState({ trackingConsentModalVisible: visible })
+  }
+
+  renderTrackingConsentItem({ item }) {
+    const isSelected = TrackingConsent[item] === this.state.trackingConsent
+    const backgroundColor = isSelected ? "#303f9f" : "#448aff";
+    const color = isSelected ? 'white' : 'black';
+
+    return (
+      <TrackingConsentItem
+        item={item}
+        onPress={() => {
+          const consent = TrackingConsent[item]
+          this.setState({ trackingConsent: consent })
+        }}
+        backgroundColor={{ backgroundColor }}
+        textColor={{ color }}
+      />
+    );
+  }
+
   render() {
     return <View style={styles.defaultScreen}>
       <Text>{this.state.welcomeMessage}</Text>
       <View style={{ marginTop: 40, alignItems: "center" }}>
-        <Text>{this.state.callDatadogButtonAction}</Text>
+        <Button
+          title={`Tracking Consent: ${this.state.trackingConsent}`}
+          accessibilityLabel="update_tracking_consent"
+          onPress={() => this.setTrackingConsentModalVisible(true)}
+        />
+        <Modal
+          animationType="slide"
+          transparent={true}
+          visible={this.state.trackingConsentModalVisible}>
+          <View style={styles.centeredView}>
+            <View style={styles.modalView}>
+              <Text style={styles.modalTitle}>Tracking Consent</Text>
+              <FlatList
+                style={{ marginTop: 20 }}
+                data={Object.keys(TrackingConsent)}
+                renderItem={this.renderTrackingConsentItem.bind(this)}
```

Comment:
I am having problems to find where this data set is being `set`. Can you give me a hint ?

Reply:
data set is static and is just a set of `TrackingConsent` enum values, and then we provide a reference to the render function which is bound to the particular scope.